test/test_autograd.py

```diff
@@ -555,6 +555,49 @@ def test_not_implemented_fwad(self):
                 # if forward AD ends up being implemented for torch.igamma, choose a different op
                 torch.igamma(dual_x, dual_x)
 
+    # def test_saved_tensor_hooks_extra_exit_during_bw_no_crash(self):
+    #     # This usage of saved tensor is not supported, but should not crash
+    #     def unpack(x):
+    #         ctx_1.__exit__()
+    #         return x
+
+    #     ctx_1 = torch.autograd.graph.saved_tensors_hooks(lambda x: x, unpack)
+    #     ctx_2 = torch.autograd.graph.saved_tensors_hooks(lambda x: x, lambda x: x)
+
+    #     for i in range(10):
+    #         with ctx_2:
+    #             ctx_1.__enter__()
+    #             x = torch.randn(3, 3, requires_grad=True)
+    #             x.sin().sum().backward()
+
+    #     # Clean up
+    #     for i in range(10):
+    #         ctx_1.__exit__()
+
+    #     # Validate there are no more hooks on the stack
+    #     a = torch.tensor(1.0, requires_grad=True)
+    #     y = a.exp()
+    #     y.grad_fn._raw_saved_result.register_hooks(lambda x: x, lambda x: x)
+
+    # def test_saved_tensor_hooks_extra_enter_during_bw_no_leak(self):
+    #     # This usage of saved tensor is not supported, but should not leak
+    #     def scope():
+    #         def unpack(x):
+    #             weak_ctx_1().__enter__()
+    #             return x
+
+    #         ctx_1 = torch.autograd.graph.saved_tensors_hooks(lambda x: x, unpack)
+    #         weak_ctx_1 = weakref.ref(ctx_1)
+
+    #         x = torch.randn(3, 3, requires_grad=True)
+    #         with ctx_1:
+    #             x.sin().sum().backward()
+    #         return weakref.ref(unpack)
+
+    #     with disable_gc():
+    #         unpack_hook_ref = scope()
+    #         self.assertIsNone(unpack_hook_ref())
+
```

Comment:
Temporarily commenting this out to see if there is some interaction with this test and the failing compiled autograd test
```
2024-07-29T21:37:39.6996815Z [31m[1m_________________ TestCompiledAutograd.test_autograd_cpp_node __________________[0m
2024-07-29T21:37:39.6997323Z Traceback (most recent call last):
2024-07-29T21:37:39.6997977Z   File "/var/lib/jenkins/workspace/test/inductor/test_compiled_autograd.py", line 1451, in test_autograd_cpp_node
2024-07-29T21:37:39.6998644Z     self.check_output_and_recompiles(fn, 2)
2024-07-29T21:37:39.6999327Z   File "/var/lib/jenkins/workspace/test/inductor/test_compiled_autograd.py", line 86, in check_output_and_recompiles
2024-07-29T21:37:39.6999998Z     actual = list(opt_fn())
2024-07-29T21:37:39.7000516Z   File "/var/lib/jenkins/workspace/test/inductor/test_compiled_autograd.py", line 1447, in fn
2024-07-29T21:37:39.7001053Z     loss.backward()
2024-07-29T21:37:39.7001638Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_tensor.py", line 522, in backward
2024-07-29T21:37:39.7002209Z     torch.autograd.backward(
2024-07-29T21:37:39.7002784Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/autograd/__init__.py", line 346, in backward
2024-07-29T21:37:39.7003394Z     _engine_run_backward(
2024-07-29T21:37:39.7004005Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/autograd/graph.py", line 806, in _engine_run_backward
2024-07-29T21:37:39.7004892Z     return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
2024-07-29T21:37:39.7005889Z RuntimeError: compiled_args not implemented for non-traceable node: torch::autograd::CppNode<CustomOpAutogradFunction>
2024-07-29T21:37:39.7006430Z 
```